File: Fuga com Helicóptero.py

number = [int(itens) for itens in input().split()]

# facilitando entendimento
helicoptero = number[0]
policia = number[1]
fugitivo = number[2]

while True:

    if fugitivo == 15 and number[3] == 1:
        fugitivo = 0
    elif fugitivo == 0 and number[3] == -1:
        fugitivo = 15   
    else:
        fugitivo += number[3]
    
    if fugitivo == policia:
            print('N')
            break
    elif fugitivo == helicoptero:
        print('S')
        break

File: other/Fuga com Helicóptero.c
#include <stdio.h>

int main(void)
{
    int helicoptero, policia, fugitivo, sentido;
    scanf("%d %d %d %d", &helicoptero, &policia, &fugitivo, &sentido);

    while (1)
    {
        if (fugitivo == 15 && sentido == 1)
            fugitivo = 0;
        else if (fugitivo == 0 && sentido == -1)
            fugitivo = 15;
        else
            fugitivo += sentido;

        if (fugitivo == helicoptero)
        {
            printf("S\n");
            break;
        }
        else if (fugitivo == policia)
        {
            printf("N\n");
            break;
        }    
    }

    return 0;
}